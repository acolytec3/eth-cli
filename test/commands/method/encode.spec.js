const MethodEncodeCommand = require('../../../src/commands/method/encode')

describe('encode', () => {
  let stdoutResult

  beforeEach(() => {
    stdoutResult = []
    jest
      .spyOn(process.stdout, 'write')
      .mockImplementation(val => stdoutResult.push(require('strip-ansi')(val.toString('utf8'))))
  })

  afterEach(() => jest.restoreAllMocks())

  it(`Should run 'encode' with empty args and flags and throw an error.`, async () => {
    await expect(MethodEncodeCommand.run()).rejects.toThrow()
  })

  it(`Should run 'encode' and throw an error.`, async () => {
    await expect(MethodEncodeCommand.run([])).rejects.toThrow(
      'Missing 2 required args:\n' +
        'abi         The abi file.\n' +
        'methodCall  e.g.: \'myMethod(arg1,arg2,["a","b",3,["d","0x123..."]])\'\n' +
        'See more help with --help'
    )
  })

  it(`Should run 'encode --help' and throw an error.`, async () => {
    await expect(MethodEncodeCommand.run(['--help'])).rejects.toThrow('EEXIT: 0')
  })

  it(`Should run 'encode --abi' and throw an error.`, async () => {
    await expect(MethodEncodeCommand.run(['LOL'])).rejects.toThrow(`Missing 1 required arg:
methodCall  e.g.: 'myMethod(arg1,arg2,["a","b",3,["d","0x123..."]])'
See more help with --help`)
  })

  it(`Should run 'encode' and get success`, async () => {
    await MethodEncodeCommand.run([
      '--sokol',
      './test/files/contracts/Proxy.abi',
      'updateAppInstance()'
    ])
    expect(stdoutResult).toMatchSnapshot()
  })
})
