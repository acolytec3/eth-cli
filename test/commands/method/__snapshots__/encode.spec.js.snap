// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encode Should run 'encode' and get success 1`] = `
Array [
  "0x3e7f54a9
",
]
`;
